app/javascript/sidebar-widget/SidebarWidget.jsx

```diff
@@ -38,58 +22,57 @@ class SidebarWidget extends Component {
     })
       .then((response) => response.json())
       .then((json) => {
-        this.setState({ suggestedUsers: json });
+        setSuggestedUsers(json);
       })
       .catch((error) => {
-        this.setState({ suggestedUsers: [] });
+        setSuggestedUsers([]);
         Honeybadger.notify(error);
       });
-  }
+  }, []);
 
-  followUser(user) {
-    const { suggestedUsers } = this.state;
-    const updatedUser = user;
-    const updatedSuggestedUsers = suggestedUsers;
-    const userIndex = suggestedUsers.indexOf(user);
+  const followUser = useCallback(
+    (user) => {
+      const updatedUser = user;
+      const updatedSuggestedUsers = suggestedUsers;
+      const userIndex = suggestedUsers.indexOf(user);
 
-    const followBtn = document.getElementById(
-      `widget-list-item__follow-button-${updatedUser.username}`,
-    );
-    followBtn.innerText = updatedUser.following ? 'Follow' : 'Following';
-
-    const toggleFollowState = (newFollowState) => {
-      updatedUser.following = newFollowState === 'followed';
-      updatedSuggestedUsers[userIndex] = updatedUser;
-      this.setState({ suggestedUsers: updatedSuggestedUsers });
-    };
-    sendFollowUser(user, toggleFollowState);
-  }
+      const followBtn = document.getElementById(
```

Comment:
Refactor (blocking): Using `useCallback` here and elsewhere in the PR seems like a premature optimization and it comes at a cost. See [When to useMemo and useCallback](https://kentcdodds.com/blog/usememo-and-usecallback).